pkgs/applications/editors/cudatext/default.nix

```diff
@@ -0,0 +1,111 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, lazarus
+, fpc
+, libX11
+
+# GTK2
+, pango
+, cairo
+, glib
+, atk
+, gtk2
+, gdk-pixbuf
+, python3
+
+# Qt5
+, libqt5pas
+, qt5
+
+, widgetset ? "qt5"
+# See https://github.com/Alexey-T/CudaText-lexers
+, additionalLexers ? [ "Nix" ]
+}:
+
+assert builtins.elem widgetset [ "gtk2" "qt5" ];
+
+let
+  deps = lib.mapAttrs
+    (name: spec:
+      fetchFromGitHub {
+        owner = "Alexey-T";
+        repo = name;
+        inherit (spec) rev sha256;
+      }
+    )
+    (builtins.fromJSON (builtins.readFile ./deps.json));
+in
+stdenv.mkDerivation rec {
+  pname = "cudatext";
+  version = "1.110.0";
+
+  src = fetchFromGitHub {
+    owner = "Alexey-T";
+    repo = "CudaText";
+    rev = version;
+    sha256 = "0mlb41q2wjlpz6myp6rhrb8bfz19357kfvvr8nsxbgzc4hspmldv";
+  };
+
+  postPatch = ''
+    substituteInPlace app/proc_globdata.pas \
+      --replace "/usr/share/cudatext" "$out/share/cudatext" \
+      --replace "libpython3.so" "${python3}/lib/libpython3.so"
+
+    # Don't check for update
+    sed -i "2138imnuHelpCheckUpd.Enabled:=false;" app/formmain.pas
```

Comment:
Isn't this just _too_ fragile? Maybe better to search for the function name and insert after it starts, or use a patch?

Reply:
OK, fixed.